src/controllers/dynakube/activegate/internal/statefulset/statefulset.go

```diff
@@ -51,6 +51,7 @@ func (statefulSetBuilder StatefulSetBuilder) getBase() appsv1.StatefulSet {
 	var sts appsv1.StatefulSet
 	sts.ObjectMeta = statefulSetBuilder.getBaseObjectMeta()
 	sts.Spec = statefulSetBuilder.getBaseSpec()
+	sts.Spec.Template.ObjectMeta.Annotations = kubeobjects.MergeMap(sts.Spec.Template.ObjectMeta.Annotations, statefulSetBuilder.dynakube.Spec.ActiveGate.Annotations)
```

Comment:
should be in `getBaseObjectMeta()` (line 68)

Reply:
As much as it makes sense to place it there, i encountered a problem. when placing it there the annotations from the dynakube seem to be overwritten and only the default annotations are there. This might be because we first get the `ObjectMeta` itself where the annotations are set to an empty map, and then we get the `Spec` where also again the `ObjectMeta` is changed and the annotations are set to the default ones.

So I think the approach from Lukas to just introduce a function for example `statefulSetBuilder.addAnnotations(&sts)` in line 54 is the best way here. What do you think?